pkg/controller/druid/handler.go

```diff
@@ -115,6 +116,15 @@ func deployDruidCluster(sdk client.Client, m *v1alpha1.Druid) error {
 			return err
 		}
 
+		if nodeSpec.PersistentVolumeClaim != nil && nodeSpec.Kind == "Deployment" {
```

Comment:
I wonder why are we restricting PVC creation to Deployment only ? I came across a case where i want to use sts with multiple pvc mounts for heap dumps. with a shared storage class. Ideally operator should support both.